### PR TITLE
Adds 0-log to Config

### LINKED ISSUE #300: Add For-Orchestrator-Failure (stderr) logger
Part of the self-healing as described in https://github.com/zero-os/0-orchestrator/blob/master/specs/selfhealing/storage.md

In certain scenarios the nbdserver has to communicate with the 0-orchestrator:

+ a tlog server crashed;
+ a storage server crashed, rebalancing of data is required;

In these cases it will do so by using a special logger, which will log the failure/success messages via stderr, which will then be picked up by the 0-watchdog and delegated over to the 0-orchestrator.

The proposed format by @FastGeert is:

```
FAILURE: {"type": "storageengine-failure", "data": "10.100.0.1:22005"}
```

and

```
SUCCESS: {"type": "storageengine-rebalance", "data": "10.100.0.1:22005"}
```

The proposed format might not be final, but it should at the very least serve as a starting point.

---

Note: as this feature is going to be used by all 0-services, we might want to move this into a more common repo, perhaps even 0-orchestrator, but for now we could simply have it as a new root module in the 0-Disk repo.

### LINKED ISSUE #363: notify 0-Orchestrator when a hot-reloaded config is invalid
Whenever a config which is hot reloaded because of an update trigged by the 0-Orchestrator, results in an invalid config for the vdisk in question, we ignore that config for usage, as it would break our mounted vdisk.

While this is fine, we do also need to notify the 0-Orchestrator, so its aware of this failure and can fix it with a correction (if possible). Make sure to add to the failure the key of the config, and any other useful information, that might allow the 0-Orchestrator to auto-fix the issue or handle it in some way.

This issue can be resolved together with #300.

### LINKED ISSUE #370: notify 0-orchestrator when etcd (config) cluster is down
The 0-orchestrator should be handled in case the etcd (config) cluster is down, as this would prevent us from updating (see: hotreloading) any watched config values (although I guess if the cluster is down they are no longer watched). But more importantly, it would prevent us from mounting a vdisk, as the configuration comes from etcd.

Sooo... We should notify 0-orchestrator about this, using the logger described in #300, such that it can make the given cluster work again, or reboot the nbdserver using a newly given config cluster.

Until 0-orchestrator has resolved this notification, no new vdisks can be mounted and no configs of active vdisks can be updated.

### PR DESCRIPTION
**Pull request by chrisvdg**
Implements 0-log to log failing config for 0-Orchestrator
closes #363 , closes #300 , closes #370

### CODE REVIEW

**Comment by chrisvdg** [comment]
Better suggestions for the naming are always welcome

File: config/logger.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+// MessageState represents the MessageSubject state
+type MessageState uint8
+
+// states
+const (
+	// StateSuccess represents a success state
+	StateSuccess MessageState = iota
+	// StateRebalanced represents a rebalanced state
+	StateRebalanced
```
Comment (GlenDC) [inline config/logger.go]: You can delete this, as this is not used anywhere for now, AFAIK.

File: config/logger.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	SubStorageengine MessageSubject = iota
+	// SubETCDCluster represents an etcd cluster subject
+	SubETCDCluster
+	// SubTlogRPC represents a Tlog RPC subject
+	SubTlogRPC
+)
+
+// subjects
+const (
+	subStorageengineStr = "storageengine"
```
Comment (GlenDC) [inline config/logger.go]: find a way to separate your words, either caps, or underscores, but choose something, and be consistent with the rest of the ecosystem.

File: config/logger.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+// logFormat represents a json struct that can be used by 0-Log
+// to notify the 0-Orchestrator of errors
+type logFormat struct {
+	Type string `json:"type"`
+	Data string `json:"data"`
+}
+
+// SendJSONLog send a logFormat formated JSON message through 0-log with provided type and data
+func SendJSONLog(t string, d string) error {
```
Comment (GlenDC) [inline config/logger.go]: you can delete this function, it's not needed.

**Comment by GlenDC** [comment]
Also why did you add this to the config module? The log module is a much better fit imho. It's about logging! Not about configuration o.o

**Pull request merged**